DBZ-1935 fixed broken links and anchors

ahus1 opened the pull request:
also added .asciidoctorconfig for improved preview and validation.
If you think the contents of .asciidoctorconfig should be in antora.yml, please let me know and I'll move them. 

I've used the latest preview version 0.30.51 of the AsciiDoc IntelliJ plugin to run "Analyze | Inspect code..." with all AsciiDoc validations. With these changes it doesn't show any errors. 

All links are fixed to my best knowledge, let me know if any more fix-ups are required.

gunnarmorling commented on documentation/.asciidoctorconfig:
```diff
@@ -0,0 +1,6 @@
+// set of configurations usually provided by in antora playbook, but
```
```suggestion
// set of configurations usually provided in Antora playbook, but
```

gunnarmorling commented:
> If you think the contents of .asciidoctorconfig should be in antora.yml, please let me know and I'll move them

So where is .asciidoctorconfig applied? Also during the Antora build? Or only in the IntelliJ preview? I general, I'd prefer to minimize the number of configuration files as much as possible, as it's quite tricky (already) to understand which is applied where and when.

ahus1 commented on documentation/modules/ROOT/pages/connectors/mysql.adoc:
```diff
@@ -14,6 +14,7 @@ The following sections provide more detailed information on how the {prodname} M
 
 * xref:assemblies/cdc-mysql-connector/as_overview-of-how-the-mysql-connector-works.adoc[Overview of how the MySQL connector works]
 * xref:assemblies/cdc-mysql-connector/as_setup-the-mysql-server.adoc[Set up MySQL Server]
+[#deploying-a-connector]
```
oops, thanks, that was a temporary state that made it into this commit

ahus1 commented:
@gunnarmorling **TL;DR:** I removed the .asciidoctorconfig and fixed some more anchors. Please re-review.

The .asciidoctorconfig is only for the preview in IntelliJ. You can use it to provide attributes that are only present in the Antora playbook.

I added it assuming that the automatically created anchors in two places (snapshots and topic-names) would require it. 
Looking closer, I found that these two attributes aren't set in the playbook, therefore the production site renders the links with the AsciiDoc defaults (that are: underscores). 
I removed .asciidoctorconfig (as it is no longer needed) and adjusted the broken anchors.

gunnarmorling commented on documentation/modules/ROOT/pages/install.adoc:
```diff
@@ -70,10 +70,10 @@ endif::[]
-xref:connectors/sqlserver.adoc#deploying-a-connector[SQL Server Connector],
+xref:connectors/mongodb.adoc#mongodb-deploying-a-connector[MongoDB Connector],
+xref:connectors/sqlserver.adoc#sqlserver-deploying-a-connector[SQL Server Connector],
 xref:connectors/oracle.adoc#deploying-a-connector[Oracle Connector],
```
@ahus1, already merged this, but one question: why did you add the prefix ("mongodb", "sqlserver") to the two above, but not these two?

ahus1 replied:
@gunnarmorling - I assume "these two" apply to oracle.adoc and db2.adoc. 

The documentation for the different connectors is inconsistent in its structure: some named the anchor with the prefix (mongodb, sqlserver), some didn't (oracle, db2, cassandra). 

What I did here: fix the broken anchors by pointing to the most appropriate existing anchor ID. What I didn't do: restructure/rename anchor IDs.

From an AsciiDoc perspective IDs need to be unique within a document. For Antora each page is a document. If you combine the documentation in a different way (maybe the downstream product?) to a big document, then it might be handy to have unique IDs across all the AsciiDoc files.

gunnarmorling replied:
>  I assume "these two" apply to oracle.adoc and db2.adoc.

Yes, exactly.

> What I did here: fix the broken anchors by pointing to the most appropriate existing anchor ID. What I didn't do: restructure/rename anchor IDs.

Ok, gotcha. Thanks for clarifying. So uniqueness within each connector documentation should be enough, we don't create one large document with all the documentation at once. Indeed I think the slight inconsistency originates from downstream work, as the docs team has added the prefix to those files which feed into the downstream product.

I.e. all good, thanks a lot again!

gunnarmorling commented:
>  I assume "these two" apply to oracle.adoc and db2.adoc.

Yes, exactly.

> What I did here: fix the broken anchors by pointing to the most appropriate existing anchor ID. What I didn't do: restructure/rename anchor IDs.

Ok, gotcha. Thanks for clarifying. So uniqueness within each connector documentation should be enough, we don't create one large document with all the documentation at once. Indeed I think the slight inconsistency originates from downstream work, as the docs team has added the prefix to those files which feed into the downstream product.

I.e. all good, thanks a lot again!
